atomics/include/desul/atomics/Macros.hpp

```diff
@@ -31,7 +32,16 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #define DESUL_HAVE_HIP_ATOMICS
 #endif
 
-#if defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__)
+#ifdef SYCL_LANGUAGE_VERSION
+#define DESUL_HAVE_SYCL_ATOMICS
+#ifdef __INTEL_LLVM_COMPILER
+#define DESUL_SYCL_NAMESPACE sycl::ONEAPI
+#else
+#define DESUL_SYCL_NAMESPACE sycl
+#endif
+#endif
```

Comment:
For me, it was a natural choice and I don't see much of a problem having it defined since it's prefixed with `DESUL_`.
Would you rather want to define and undefined it in all the files that use it?

Reply:
Why a macro rather than a namespace alias or a namespace?  I was expecting something more like:

```c++
#ifdef __INTEL_LLVM_COMPILER
namespace __sycl_atomic = ::sycl::ONEAPI
#else
namespace __sycl_atomic = ::sycl
#endif
```

Or an alias for the atomic_ref if that's all that gets pulled in.

Reply:
Sure, I'll give it a shot.